src/network/clientpackethandler.cpp

```diff
@@ -1068,14 +1070,19 @@ void Client::handleCommand_HudAdd(NetworkPacket* pkt)
 
 void Client::handleCommand_HudRemove(NetworkPacket* pkt)
 {
-	u32 id;
+	u32 server_id;
 
-	*pkt >> id;
+	*pkt >> server_id;
 
-	ClientEvent *event = new ClientEvent();
-	event->type     = CE_HUDRM;
-	event->hudrm.id = id;
-	m_client_event_queue.push(event);
+	std::unordered_map<u32, u32>::const_iterator i = m_hud_server_to_client.find(server_id);
+	if (i != m_hud_server_to_client.end()) {
+		int client_id = i->second;
+
+		ClientEvent *event = new ClientEvent();
+		event->type     = CE_HUDRM;
+		event->hudrm.id = client_id;
+		m_client_event_queue.push(event);
```

Comment:
Also remove this ID from `m_hud_server_to_client` when `Game::handleClientEvent_HudRemove` is called later.

Reply:
done